src/Notifier.cpp

```diff
@@ -163,9 +183,9 @@ size_t Notifier::ReceiveConfirmation() {
  * Reset the Shotgun-Alien queue to nullptr
  */
 void Notifier::Reset() {
-   std::lock_guard<std::mutex> guard(gLock);
    gQueue.reset(nullptr);
```

Comment:
My suggestion is to forget to make this a multi-threaded class. Use it only in a single-thread context. 

I like it that you are removing the global, file scope members and making them object members. The mutex usage in this file can likely (if single threaded use only) be completely removed. 

Please do a search and remove the other instances of where it is used. 

---- if however, this is for a multi threaded scenario then you need to explain why first ---